pkg/acs/calacs/acsccd/blev_funcs_postsm4.c

```diff
@@ -36,152 +39,218 @@ static int make_amp_array(const int arr_rows, const int arr_cols, SingleGroup *
 static int unmake_amp_array(const int arr_rows, const int arr_cols, SingleGroup * im,
                             int amp, double * array);
 
+/* Remove the signal dependent bias shift from post-SM4 full frame WFC data
+ * (based on ISR http://www.stsci.edu/hst/acs/documents/isrs/isr1202.pdf),
+ * as well as from supported subarray data (~post-mid-2016) as identified by
+ * specific aperture names 
+ * (based on ISR http://www.stsci.edu/hst/acs/documents/isrs/isr1703.pdf).
+ *
+ * chip2 consists of amps C & D, chip1 consists of amps A & B. 
+ *
+ * Because full frame data uses both chips, but supported subarray data
+ * is only from a single amp (therefore, a portion of a single chip), this
+ * routine has a variable number of input SingleGroups.
+ *
+ * 16-May-2018 M.D. De La Pena: Generalized bias_shift_corr() to handle subarray data.
+ */
+int bias_shift_corr(ACSInfo *acs, int nGroups, ...) {
+
+  /* Code to handle variable number of input parameters.  The number of SingleGroup is 2 when 
+     processing a full frame (2 chips - all 4 amps) and only 1 when processing a supported subarray 
+     (1 chip from a single amp).
+  */
+  SingleGroup *sg[nGroups];
+  va_list arguments;
+
+  /* Initializing arguments to store all the values being passed to this routine via "..." */
+  va_start (arguments, nGroups);
+  {
```

Comment:
Why the extra scope?

Reply:
Perhaps you want to scope ``arguments``, which is a very good idea, in which case this needs to be moved up just above `` va_list arguments;`` and the closing, just after ``va_end (arguments);``.

Reply:
If that was not your intention, I would scope ``arguments``.